### PR TITLE
hw-29

### PR DESCRIPTION
**Pull request by julia-shvabovich**

### CODE REVIEW

File: src/main/java/application/controller/IndexController.java [hunk 1]
```diff
@@ -9,9 +14,19 @@
...
+    private void addAdmins() {
+        User admin = new User("admin", "admin", "12345");
+        admin.setRoles(Set.of(Role.of("ADMIN")));
+        userService.create(admin);
+    }
+
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        addAdmins();
```
Comment (tarkaiev) [inline src/main/java/application/controller/IndexController.java]: we will try to add admin every time we reach index page?

File: src/main/java/application/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -25,8 +25,12 @@ public void doFilter(ServletRequest servletRequest,
             throws IOException, ServletException {
         HttpServletRequest req = (HttpServletRequest) servletRequest;
         HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        Set<String> availablePages = new HashSet<>();
+        availablePages.add("/login");
```
Comment (tarkaiev) [inline src/main/java/application/web/filter/AuthenticationFilter.java]: we can do it in init() method

File: src/main/java/application/controller/IndexController.java (hunk 1 above)
Comment (julia-shvabovich) [reply src/main/java/application/controller/IndexController.java]: done

File: src/main/java/application/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package application.model;
+
+public class Role {
+    private Long id;
+    private RoleName roleName;
+
+    public Role(RoleName roleName) {
```
Comment (KseniiaMakarova) [inline src/main/java/application/model/Role.java]:
```suggestion
    private Role(RoleName roleName) {
```

File: src/main/webapp/WEB-INF/view/access-denied.jsp [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+<head>
+    <title>Access denied</title>
+</head>
+<body>
+<%@include file="header.jsp"%>
+<br>
+<h1>Sorry! Access to this page is denied</h1>
+<br>
+<center>
+    <img style="max-width: 300px; height: inherit" src="https://i.imgur.com/WxXk1T5.png" alt="garold">
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/view/access-denied.jsp:12]: nice

File: src/main/java/application/web/filter/AuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String requestedUrl = req.getServletPath();
+        if (protectedUrls.get(requestedUrl) == null) {
```
Comment (KseniiaMakarova) [inline src/main/java/application/web/filter/AuthorizationFilter.java]: let's put both conditions in one `if`
also, we can use `containsKey()` here

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**